File: dsl-style/src/test/java/example/dsl_style/BatchDeleteTest.java
package example.dsl_style;

import example.dsl_style.repository.EmployeeRepository;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.seasar.doma.jdbc.Config;

@ExtendWith(TestEnvironment.class)
public class BatchDeleteTest {

  private final EmployeeRepository repository;

  BatchDeleteTest(Config config) {
    repository = new EmployeeRepository(config);
  }

  @Test
  public void testBatchDelete() {
    var list = repository.selectAll();
    repository.batchDelete(list);
  }
}

File: dsl-style-kotlin/src/test/kotlin/example/dsl_style_kotlin/BatchDeleteTest.kt
package example.dsl_style_kotlin

import example.dsl_style_kotlin.repository.EmployeeRepository
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.seasar.doma.jdbc.Config

@ExtendWith(TestEnvironment::class)
class BatchDeleteTest internal constructor(config: Config) {

    private val repository: EmployeeRepository = EmployeeRepository(config)

    @Test
    fun testBatchDelete() {
        val list = repository.selectAll()
        repository.batchDelete(list)
    }
}
